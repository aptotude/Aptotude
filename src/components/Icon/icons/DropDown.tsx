import React from 'react';

const SvgDropDown = (props: React.SVGProps<SVGSVGElement>) => (
  <svg viewBox="0 0 24 24" {...props}>
    <path d="M7 10l5 5 5-5z" />
    <path d="M0 0h24v24H0z" fill="none" />
  </svg>
);

export default SvgDropDown;
