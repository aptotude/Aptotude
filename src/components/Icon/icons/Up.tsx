import React from 'react';

const SvgUp = (props: React.SVGProps<SVGSVGElement>) => (
  <svg viewBox="0 0 24 24" {...props}>
    <path d="M12 8l-6 6 1.41 1.41L12 10.83l4.59 4.58L18 14z" />
    <path d="M0 0h24v24H0z" fill="none" />
  </svg>
);

export default SvgUp;
