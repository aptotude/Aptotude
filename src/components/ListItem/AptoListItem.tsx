import React from 'react';
import classNames from 'classnames';
import './aptoListItem.scss';
import { StandardTypes } from '../../utils/standardTypes';

interface Props extends StandardTypes {
  type?: 'li' | null | undefined;
  link?: boolean;
  active?: boolean;
}

const COMPONENT_PREFIX = 'AptoListItem';

export default class AptoListItem extends React.Component<Props> {
  public static defaultProps = {
    link: null
  };

  public render() {
    const {
      type,
      className,
      forwardRef,
      children,
      link,
      active,
      ...rest
    } = this.props;

    const classes = classNames(
      COMPONENT_PREFIX,
      type ? `${COMPONENT_PREFIX}--${type}` : undefined,
      link && `${COMPONENT_PREFIX}--link`,
      active && `${COMPONENT_PREFIX}--active`,
      className
    );

    const Component = type || 'div';

    return (
      <Component
        {...rest}
        ref={forwardRef}
        className={classes}>
          <div className="AptoListItem-content">
            <div className="AptoListItem-text">
              {children}
            </div>
          </div>
      </Component>
    );
  }
}
