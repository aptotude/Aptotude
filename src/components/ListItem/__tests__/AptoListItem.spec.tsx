import * as React from 'react';
import { render, cleanup, fireEvent, getByText } from 'react-testing-library';
import { AptoListItem } from '../AptoListItem';
import { person } from '@apto/icons';

afterEach(() => {
  cleanup();
});

describe('AptoListItem', () => {
  it('Renders', () => {
    const { container } = render(<AptoListItem>hi</AptoListItem>);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('Renders link', () => {
    const { container } = render(<AptoListItem link>hi</AptoListItem>);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('Renders empty', () => {
    const { container } = render(
      <AptoListItem empty>nothing found</AptoListItem>
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('Renders active', () => {
    const { container } = render(<AptoListItem active>active</AptoListItem>);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders extra class', () => {
    const { container } = render(
      <AptoListItem className="fooo">extra classes</AptoListItem>
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('clicks', () => {
    let called = false;
    const { container } = render(
      <AptoListItem
        onClick={() => {
          called = true;
        }}
      >
        clicky
      </AptoListItem>
    );
    fireEvent.click(getByText(container, 'clicky'));
    expect(called).toBeTruthy();
  });

  it('Renders icon', () => {
    const { container } = render(<AptoListItem icon={person}>hi</AptoListItem>);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('Renders icon options', () => {
    const { container } = render(
      <AptoListItem
        iconSize="5"
        iconColor="blue"
        iconStatus="warning"
        icon={person}
      >
        hi
      </AptoListItem>
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('Renders circle icon options', () => {
    const { container } = render(
      <AptoListItem
        iconSize="5"
        iconCircleColor="blue"
        iconCircle
        iconStatus="warning"
        icon={person}
      >
        hi
      </AptoListItem>
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
