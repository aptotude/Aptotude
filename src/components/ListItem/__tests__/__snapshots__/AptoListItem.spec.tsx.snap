// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AptoListItem Renders 1`] = `
<div
  class="AptoListItem"
>
  <div
    class="AptoListItem-content"
  >
    hi
  </div>
</div>
`;

exports[`AptoListItem Renders active 1`] = `
<div
  class="AptoListItem AptoListItem--active"
>
  <div
    class="AptoListItem-content"
  >
    active
  </div>
</div>
`;

exports[`AptoListItem Renders circle icon options 1`] = `
<div
  class="AptoListItem AptoListItem--icon"
>
  <div
    class="AptoListItem-iconWrapper"
  >
    <span
      class="AptoIcon AptoIcon--size5 AptoIcon--inline AptoIcon--circle AptoIcon--statusWarning AptoIcon--circleColorBlue"
    >
      <svg
        focusable="false"
        height="100%"
        preserveAspectRatio="xMidYMid meet"
        viewBox="0 0 24 24"
        width="100%"
      >
        <path
          d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
        />
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
      </svg>
      <span
        class="AptoIcon-status"
      >
        !
      </span>
    </span>
  </div>
  <div
    class="AptoListItem-content"
  >
    hi
  </div>
</div>
`;

exports[`AptoListItem Renders empty 1`] = `
<div
  class="AptoListItem AptoListItem--empty"
>
  <div
    class="AptoListItem-content"
  >
    nothing found
  </div>
</div>
`;

exports[`AptoListItem Renders icon 1`] = `
<div
  class="AptoListItem AptoListItem--icon"
>
  <div
    class="AptoListItem-iconWrapper"
  >
    <span
      class="AptoIcon AptoIcon--size3 AptoIcon--inline"
    >
      <svg
        focusable="false"
        height="100%"
        preserveAspectRatio="xMidYMid meet"
        viewBox="0 0 24 24"
        width="100%"
      >
        <path
          d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
        />
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
      </svg>
    </span>
  </div>
  <div
    class="AptoListItem-content"
  >
    hi
  </div>
</div>
`;

exports[`AptoListItem Renders icon options 1`] = `
<div
  class="AptoListItem AptoListItem--icon"
>
  <div
    class="AptoListItem-iconWrapper"
  >
    <span
      class="AptoIcon AptoIcon--size5 AptoIcon--inline AptoIcon--statusWarning AptoIcon--colorBlue"
    >
      <svg
        focusable="false"
        height="100%"
        preserveAspectRatio="xMidYMid meet"
        viewBox="0 0 24 24"
        width="100%"
      >
        <path
          d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
        />
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
      </svg>
      <span
        class="AptoIcon-status"
      >
        !
      </span>
    </span>
  </div>
  <div
    class="AptoListItem-content"
  >
    hi
  </div>
</div>
`;

exports[`AptoListItem Renders link 1`] = `
<div
  class="AptoListItem AptoListItem--link"
>
  <div
    class="AptoListItem-content"
  >
    hi
  </div>
</div>
`;

exports[`AptoListItem renders extra class 1`] = `
<div
  class="AptoListItem fooo"
>
  <div
    class="AptoListItem-content"
  >
    extra classes
  </div>
</div>
`;
